jni/src/Foreign/JNI.hs

```diff
@@ -360,25 +389,35 @@ getStaticFieldID
   -> JNI.String -- ^ JNI signature
   -> IO JFieldID
 getStaticFieldID cls fieldname sig = withJNIEnv $ \env ->
+    withJ cls $ \clsp ->
     throwIfException env $
     JNI.withString fieldname $ \fieldnamep ->
     JNI.withString sig $ \sigp ->
     [CU.exp| jfieldID {
       (*$(JNIEnv *env))->GetStaticFieldID($(JNIEnv *env),
-                                          $(jclass cls),
+                                          $(jclass clsp),
                                           $(char *fieldnamep),
                                           $(char *sigp)) } |]
 
+getObjectField :: Coercible o (J a) => o -> JFieldID -> IO JObject;
+getObjectField (coerce -> upcast -> obj) field = withJNIEnv $ \env ->
```

Comment:
It has a call to `newJ` which the other calls haven't. Same for `getStaticObjectField`.